src/main/java/org/sonar/plugins/stash/PluginUtils.java

```diff
@@ -39,9 +47,42 @@ public static PluginInfo infoForPluginClass(Class klass) {
                 return new PluginInfo(pluginName, pluginVersion);
             }
         } catch (IOException e) {
+            LOGGER.warn(ERROR_DETAILS, e.getMessage());
+            LOGGER.debug(ERROR_STACK, e);
             return null;
         }
 
         return null;
     }
+
+    private static boolean classesMatching(String className, Class right) {
+
+        boolean doesMatch = false; 
+
+        // Let's compare the plugin's class with another one (without '==', see SQUID:S1872)
+        try {
+            Class<?> left   = Class.forName(className);
+            Object leftObj  = left.getClass().newInstance();
+            Object rightObj = right.getClass().newInstance();
+        
+            if (  left.isInstance(rightObj)
+             ||  right.isInstance(leftObj) ) {
+                doesMatch = true;
+            }
+
+        } catch (ClassNotFoundException e) {
+            LOGGER.warn(ERROR_DETAILS, e.getMessage());
+            LOGGER.debug(ERROR_STACK, e);
+
+        } catch (InstantiationException e) {
+            LOGGER.warn(ERROR_DETAILS, e.getMessage());
+            LOGGER.debug(ERROR_STACK, e);
+
+        } catch (IllegalAccessException e) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Combine this catch with the one at line 73, which has the same body. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2147)